Library/Homebrew/dev-cmd/update-python-resources.rb

```diff
@@ -37,6 +37,10 @@ def update_python_resources_args
   def update_python_resources
     args = update_python_resources_args.parse
 
+    # Python packages may need access to additional commands while generating pip report.
```

Comment:
Which commands, for example?

Reply:
Off the top of my head, I recall seeing `pkg-config` and `msgfmt` (from gettext).

Reply:
I get `setup.py` is a script, but why are things invoking tools on a basic metadata parse? Is it just bad scripts calling things outside of build steps?

Reply:
> EDIT: Alternatively, we could go with explicit list and do something like Superenv to only add those `opt_bin` dirs. I went with simpler option as we already are using this `PATH` in `brew bump` / `brew bump-formula-pr`.

Yeh, I think this might be nicer.

Reply:
> I get `setup.py` is a script, but why are things invoking tools on a basic metadata parse? Is it just bad scripts calling things outside of build steps?

Pretty much. One of the issues of having metadata and build commands in the same file as it can bleed over.

---

> Are the failures obvious?

Usually. Though via `brew` the errors are hidden without `--verbose`.
```console
❯ brew update-python-resources borgbackup
==> Retrieving PyPI dependencies for "borgbackup==1.2.7"...
Error: Unable to determine dependencies for "borgbackup==1.2.7" because of a failure when running
`/opt/homebrew/opt/python@3.12/libexec/bin/python -m pip install -q --disable-pip-version-check --dry-run --ignore-installed --report=/dev/stdout borgbackup==1.2.7`.
Please update the resources manually.
```
After verbose, the error at end is:
```console
      OSError: pkg-config probably not installed: FileNotFoundError(2, 'No such file or directory')
```

---

> Having at least a couple documented examples here might be good.

Sure, if I keep going down this route.

---

> > EDIT: Alternatively, we could go with explicit list and do something like Superenv to only add those `opt_bin` dirs. I went with simpler option as we already are using this `PATH` in `brew bump` / `brew bump-formula-pr`.
> 
> Yeh, I think this might be nicer.

When I have a chance, I'll explore this option further. Logic will most likely reside in `utils/pypi.rb`.

This should also help run `brew bump` which may allow us to autobump these (so any extra commands are done in CI environment rather than on user system).

Reply:
Cool, thanks for the example. `borgbackup` looks like it might improve with 1.4.0b1 but I recognise the need for a broader solution here.